lib/rspec/mocks/any_instance/recorder.rb

```diff
@@ -18,6 +18,9 @@ def initialize(klass)
           @stubs = Hash.new { |hash, key| hash[key] = [] }
           @observed_methods = []
           @played_methods = {}
+          @own_methods = (klass.public_instance_methods(false) +
+                         klass.protected_instance_methods(false) +
+                         klass.private_instance_methods(false)).map(&:to_sym)
```

Comment:
Since a force push will wipe this out... putting @myronmarston's comments here so the "outdated" diff will still match up with his comments. 

> Thanks for taking a stab at this, @jrafanie. I'm a bit concerned about the performance implications of your solution here -- mapping over all methods (e.g. map(&:to_sym)) is a linear operation that we'd like to avoid (particularly because the sorts of legacy systems where users use any_instance tend to be ones where there objects have very large interfaces). Likewise, the include? check is a linear operation as well.
Did an owner check not work? That route seems simpler and likely to perform better.